aikau/src/main/resources/alfresco/forms/controls/Tree.js

```diff
@@ -0,0 +1,179 @@
+ * allow users to select a value from within a tree like structure. In order to popuplate the tree
+ * it is necessary to set the "optionsConfig.publishTopic" configuration to publish a topic that
+ * will result in data being requested for the children of each tree node expanded.</p>
+ * <p>The value of the control will be either the selected object or the property of the selected
+ * item defined by the [valueProperty]{@link module:alfresco/forms/controls/Tree#valueProperty}</p>
+ * 
+ * @example <caption>Example configuration</caption>
+ * {
+ *   id: "CLOUD_SYNC_CONTAINER",
+ *   name: "alfresco/forms/controls/Tree",
+ *   config: {
+ *     fieldId: "CLOUD_SYNC_CONTAINER",
+ *     name: "targetFolderNodeRef",
+ *     label: "cloud-sync.dialog.path.label",
+ *     valueProperty: "nodeRef",
+ *     optionsConfig: {
+ *        publishTopic: topics.GET_FORM_VALUE_DEPENDANT_OPTIONS,
+ *        publishPayload: {
+ *           publishTopic: topics.GET_CLOUD_PATH
+ *        },
+ *        publishGlobal: false
+ *     },
+ *   }
+ * },
+ * 
+ * @module alfresco/forms/controls/Tree
+ * @extends module:alfresco/forms/controls/BaseFormControl
+ * @author Dave Draper
+ * @since 1.0.39
+ */
+define(["dojo/_base/declare",
+        "alfresco/forms/controls/BaseFormControl",
+        "alfresco/navigation/Tree",
+        "dojo/_base/lang"], 
+        function(declare, BaseFormControl, Tree, lang) {
+   
+   return declare([BaseFormControl], {
+      
+      /**
+       * The value to retrieve from the selected tree node. If left as null then the entire
+       * selected object will be used as the value.
+       * 
+       * @instance
+       * @type {string}
+       * @default
+       */
+      valueProperty: null,
+
+      /**
+       * @instance
+       */
+      getWidgetConfig: function alfresco_forms_controls_Tree__getWidgetConfig() {
+         return {
+            id : this.generateUuid(),
+            name: this.name
+         };
+      },
+      
+      /**
+       * Sets up the subscriptions to handling tree node selection and then calls 
+       * [createTree]{@link module:alfresco/forms/controls/Tree#createTree} to create the tree to 
+       * be used as the form control.
+       * 
+       * @instance
+       * @param {object} config The configuration to use when instantiating the form control
+       */
+      createFormControl: function alfresco_forms_controls_Tree__createFormControl(/*jshint unused:false*/ config) {
+         this._treeNodeSelectedTopic = this.generateUuid();
+         this.alfSubscribe(this._treeNodeSelectedTopic, lang.hitch(this, this.onTreeNodeSelected));
+         return this.createTree();
+      },
+
+      /**
+       * This creates a new [tree]{@link module:alfresco/navigation/Tree}.
+       * 
+       * @instance
+       * @return {object} A new tree instance
+       */
+      createTree: function alfresco_forms_controls_Tree__createTree() {
+         return new Tree({
```

Comment:
Not using `this.createWidget` intentionally? Would negate the need to manually specify the `pubSubScope` ...